api/clients/v2/payload_disperser.go

```diff
@@ -0,0 +1,222 @@
+	nonSignerStakesAndSignature, err := pd.certVerifier.GetNonSignerStakesAndSignature(
+		timeoutCtx, blobStatusReply.GetSignedBatch())
+	if err != nil {
+		return nil, fmt.Errorf("get non signer stake and signature: %w", err)
+	}
+	pd.logger.Debug("Retrieved NonSignerStakesAndSignature", "blobKey", blobKey)
+
+	eigenDACert, err := verification.BuildEigenDACert(
+		blobStatusReply.GetBlobInclusionInfo(),
+		blobStatusReply.GetSignedBatch().GetHeader(),
+		nonSignerStakesAndSignature)
+	if err != nil {
+		return nil, fmt.Errorf("build eigen da cert: %w", err)
+	}
+	pd.logger.Debug("Constructed EigenDACert", "blobKey", blobKey)
+
+	timeoutCtx, cancel = context.WithTimeout(ctx, pd.config.ContractCallTimeout)
+	defer cancel()
+	err = pd.certVerifier.VerifyCertV2(timeoutCtx, eigenDACert)
+	if err != nil {
+		return nil, fmt.Errorf("verify cert for blobKey %v: %w", blobKey, err)
+	}
+	pd.logger.Debug("EigenDACert verified", "blobKey", blobKey)
+
+	return eigenDACert, nil
+}
+
+// Close is responsible for calling close on all internal clients. This method will do its best to close all internal
+// clients, even if some closes fail.
+//
+// Any and all errors returned from closing internal clients will be joined and returned.
+//
+// This method should only be called once.
+func (pd *PayloadDisperser) Close() error {
+	err := pd.disperserClient.Close()
+	if err != nil {
+		return fmt.Errorf("close disperser client: %w", err)
+	}
+
+	return nil
+}
+
+// pollBlobStatus polls the disperser for the status of a blob that has been dispersed
+//
+// This method will only return a non-nil BlobStatusReply if the blob is reported to be CERTIFIED prior to the timeout.
+// In all other cases, this method will return a nil BlobStatusReply, along with an error describing the failure.
+func (pd *PayloadDisperser) pollBlobStatus(
+	ctx context.Context,
+	blobKey core.BlobKey,
+	initialStatus dispgrpc.BlobStatus,
+) (*dispgrpc.BlobStatusReply, error) {
+
+	previousStatus := initialStatus
+
+	ticker := time.NewTicker(pd.config.BlobStatusPollInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf(
+				"timed out waiting for %v blob status, final status was %v: %w",
+				dispgrpc.BlobStatus_CERTIFIED,
+				previousStatus,
+				ctx.Err())
+		case <-ticker.C:
+			// This call to the disperser doesn't have a dedicated timeout configured.
+			// If this call fails to return in a timely fashion, the timeout configured for the poll loop will trigger
+			blobStatusReply, err := pd.disperserClient.GetBlobStatus(ctx, blobKey)
+			if err != nil {
+				pd.logger.Warn("get blob status", "err", err, "blobKey", blobKey)
+				continue
+			}
+
+			newStatus := blobStatusReply.Status
+			if newStatus != previousStatus {
+				pd.logger.Debug(
+					"Blob status changed",
+					"blob key", blobKey,
+					"previous status", previousStatus,
+					"new status", newStatus)
+				previousStatus = newStatus
+			}
+
+			switch newStatus {
```

Comment:
let's add a TODO specifying that we'll need to add more in-depth response status processing to derive failover errors

Reply:
I'm not sure I understand what you mean by "derive failover errors".

Do you mean we will need to expand the types of status returned from the top level `SendPayload` method, beyond what's included in the existing `BlobStatus` enum?

Reply:
Hmm. Is it not true that _any_ error in dispersal would require resorting to failover? If not, what class of error would you expect to *not* result in failover, and what would be the strategy to handle it? Retries? (trying to understand likely outcomes, to be able to write a helpful TODO)

Reply:
Added the TODO